packages/liveblocks-react/src/factory.tsx

```diff
@@ -843,15 +843,18 @@ export function createRoomContext<
 
   const commentsErrorEventSource =
     makeEventSource<CommentsApiError<TThreadMetadata>>();
-  const commentsRooms = new Map<string, CommentsRoom<TThreadMetadata>>();
+  const commentsRooms = new Map<
+    Room<TPresence, TStorage, TUserMeta, TRoomEvent>,
+    CommentsRoom<TThreadMetadata>
+  >();
 
   function getCommentsRoom(
     room: Room<TPresence, TStorage, TUserMeta, TRoomEvent>
   ) {
-    let commentsRoom = commentsRooms.get(room.id);
+    let commentsRoom = commentsRooms.get(room);
     if (commentsRoom === undefined) {
       commentsRoom = createCommentsRoom(room, commentsErrorEventSource);
-      commentsRooms.set(room.id, commentsRoom);
+      commentsRooms.set(room, commentsRoom);
```

Comment:
`commentsRooms` map now uses the room reference instead of the room id to identify a comment room. This ensures that whenever the room reference changes, a new comment room is created. 

This wasn't true when the room id was used to identify a comment room. Say, you had a room with id `nextjs-threads`, and you deleted this room and created a new room with the same id. Because the id hadn't changed between the two rooms, our comment room would still references the old room (that was destroyed).